models/sar_siames/create_embedding_DB.py

```diff
@@ -0,0 +1,76 @@
+import os
+import argparse
+import numpy as np
+import tensorflow as tf
+import pickle
+import logging
+from tensorflow.keras.models import load_model
+from sar_siames_network import l2_norm
+import keras
+keras.config.enable_unsafe_deserialization()
+
+
+def extract_label_from_filename(filename):
+    """
+    Labels: 1 if '_A.npy' (anomalous), else 0 (normal).
+    """
+    return 1 if filename.endswith("_A.npy") else 0
+
+
+def build_db(patch_folder, model_path, db_output_path):
+    logging.info(f"Loading model from {model_path}")
+    model = load_model(model_path, custom_objects={"l2_norm": l2_norm})
+
+    patch_paths = [
+        os.path.join(patch_folder, f) 
+        for f in os.listdir(patch_folder) 
+        if f.endswith(".npy")
+    ]
+    logging.info(f"Found {len(patch_paths)} patches to process")
+
+    embeddings = []
+    labels = []
+
+    for path in patch_paths:
+        patch = np.load(path)
+        patch = np.expand_dims(patch, axis=0)  # Add batch dimension
+        embedding = model.predict(patch, verbose=0)
+        label = extract_label_from_filename(os.path.basename(path))
+
+        embeddings.append(embedding[0])
+        labels.append(label)
```

Comment:
Consider extracting the loop body into function/s and looping implicitly, using list comprehension or distributing using`multiprocess`
```suggestion
    patch_embedding_prediction = lambda path: model.predict(np.expand_dims(np.load(path), axis=0), verbose=0)[0]
    embeddings = [map(patch_embedding_prediction, patch_paths)]
labels = [
    extract_label_from_filename(os.path.basename(path))
    for path in patch_paths
]
```

Reply:
updated